docs/proposals/policy.md

```diff
@@ -0,0 +1,264 @@
+	"namespace": "chisel",
+	"roleRef": {
+		"namespace": "chisel",
+		"name": "fatFingeredEditor"
+	}
+	"userNames": ["Curt"]
+}
+```
+
+## Basic Concepts
+ 1.  A PolicyRule expresses a permission containing: Deny, []Verb, []ResourceKinds (pod, deploymentConfig, *-(all kinds), etc)
+ 1.  A Role is a way to name a set of PolicyRules
+ 1.  A RoleBinding is a way to associate a single PolicyRule with a given user or team.  A RoleBinding references (but does not include) a Role in either the local namespace or the global namespace.
+ 1.  A global namespace exists.  This namespace is special.  Roles defined in the global namespace can be referenced from any other namespace.  Bindings defined in the global namespace apply to **all** namespaces (an admin in the global namespace can edit any resource in any namespace).  RoleBinding policy rules cannot be overridden by any local binding (a local PolicyBindingRule that denies actions to the global admin will be overriden by the binding defined in the global namespace).
+ 1.  A Policy is a container object that attempts to mitigate referential integrity problems.  Referential integrity problems **still exist!**  Local RoleBindings can still reference global Roles stored in a different object, so dangling references can still happen.  In usage, a RoleBinding that references a missing Role will produce an error.
+ 1.  Roles versus Teams.  A Role is a grouping of PolicyRules.  A team is a grouping of users.  A RoleBinding associates a Role with a set of users and/or teams.  This distinction makes Teams easier to share across projects.
+
+Policy Evaluation
+In order to determine whether a request is authorized, the AuthorizationAttributes are tested in the following order:
+  1. all deny RoleBinding PolicyRules in the global namespace
+  1. all allow RoleBinding PolicyRules in the global namespace
+  1. all deny RoleBinding PolicyRules in the namespace
+  1. all allow RoleBinding PolicyRules in the namespace
+
+### Configtime Authorization Types
+These are the types used in the policy example above.  They allow us: to quickly find the policy rules that apply to namespace in etcd, to easily delegate project level control to a project admin, and to easily segregate all role control (a project admin can create/update/delete RoleBindings, but not Roles).
+```
+// PolicyRule holds information that describes a policy rule, but does not contain information 
+// about who the rule applies to or which namespace the rule applies to.
+type PolicyRule struct {
+	// Deny is true if any request matching this rule should be denied.  If false, any request matching this rule is allowed.
+	Deny bool
+	// Verbs is a list of Verbs that apply to ALL the ResourceKinds and AttributeRestrictions contained in this rule.  "*" represents all kinds.
+	Verbs []Verb
+	// AttributeRestrictions will vary depending on what the Authorizer/AuthorizationAttributeBuilder pair supports.  
+	AttributeRestrictions runtime.EmbeddedObject
```

Comment:
Should define behavior when attributerestrictions is not recognized by the authorizer. I think we would have to treat that like a deny.